test/sai_test/sai_utils.py

```diff
@@ -178,3 +178,50 @@ def generate_ip_address_list(role, group, indexes):
     for index in indexes:
         ip_list.append(role.format(group, index))
     return ip_list
+
+def warm_test(is_runTest:bool=False, time_out=60, interval=1):
+    """
+    Method decorator for the method on warm testing.
+    
+    Depends on parameters [test_reboot_mode] and [test_reboot_stage].
+    Runs different method, test_starting, setUp_post_start and runTest
+    """
+    def _check_run_case(f):
+        def test_director(inst, *args):
+            if inst.test_reboot_mode == 'warm':
+                print("shutdown the swich in warm mode")
+                sai_thrift_set_switch_attribute(inst.client, restart_warm=True)
+                sai_thrift_set_switch_attribute(inst.client, pre_shutdown=True)
+                sai_thrift_remove_switch(inst.client)
+			#check and create file if not exist
+			# msg to promote change the text file content to rebooted
+                warm_file = open('/tmp/warm_reboot','w+')
+				# write content to reboot-requested
+                times = 0
+                print("write rebooting to file")
+                warm_file.write(WARM_TEST_REBOOTING)
+                warm_file.close()
+                try:
+                    while 1:
+                        warm_file = open('/tmp/warm_reboot','r')
```

Comment:
please print message when entring into each condition, with the reason and what's the effect.